pkg/antctl/raw/traceflow/command_test.go

```diff
@@ -153,3 +225,408 @@ func TestParseFlow(t *testing.T) {
+							Observations: []v1alpha1.Observation{
+								{
+									Component: v1alpha1.ComponentSpoofGuard,
+									Action:    v1alpha1.ActionForwarded,
+								},
+								{
+									Component: v1alpha1.ComponentForwarding,
+									Action:    v1alpha1.ActionForwardedOutOfOverlay,
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: `{
+  "name": "dummy-traceflow-pod-to-IP",
+  "phase": "Succeeded",
+  "source": "default/pod-1",
+  "destination": "1.1.1.1",
+  "results": [
+    {
+      "node": "node1",
+      "observations": [
+        {
+          "component": "SpoofGuard",
+          "action": "Forwarded"
+        },
+        {
+          "component": "Forwarding",
+          "action": "ForwardedOutOfOverlay"
+        }
+      ]
+    }
+  ]
+}`,
+		},
+		{
+			name:        "dummy-traceflow-livetraffic",
+			src:         srcPod,
+			liveTraffic: true,
+			outputType:  "yaml",
+			tf: v1alpha1.Traceflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dummy-traceflow-livetraffic",
+				},
+				Spec: v1alpha1.TraceflowSpec{
+					Source: v1alpha1.Source{
+						Namespace: "default",
+						Pod:       "pod-1",
+					},
+					Packet: v1alpha1.Packet{
+						SrcIP: "10.244.1.2",
+						IPHeader: v1alpha1.IPHeader{
+							Protocol: v1alpha1.ICMPProtocolNumber,
+						},
+					},
+					LiveTraffic: true,
+				},
+				Status: v1alpha1.TraceflowStatus{
+					Phase: v1alpha1.Succeeded,
+					Results: []v1alpha1.NodeResult{
+						{
+							Node: "node1",
+							Observations: []v1alpha1.Observation{
+								{
+									Component: v1alpha1.ComponentSpoofGuard,
+									Action:    v1alpha1.ActionForwarded,
+								},
+								{
+									Component: v1alpha1.ComponentForwarding,
+									Action:    v1alpha1.ActionForwardedOutOfOverlay,
+								},
+							},
+						},
+					},
+					CapturedPacket: &v1alpha1.Packet{
+						SrcIP:  "10.244.1.2",
+						DstIP:  "142.250.182.78",
+						Length: 84,
+						IPHeader: v1alpha1.IPHeader{
+							Protocol: v1alpha1.ICMPProtocolNumber,
+							TTL:      64,
+							Flags:    2,
+						},
+						TransportHeader: v1alpha1.TransportHeader{
+							ICMP: &v1alpha1.ICMPEchoRequestHeader{
+								ID:       29304,
+								Sequence: 1,
+							},
+						},
+					},
+				},
+			},
+			expected: `name: dummy-traceflow-livetraffic
+phase: Succeeded
+source: default/pod-1
+results:
+- node: node1
+  observations:
+  - component: SpoofGuard
+    action: Forwarded
+  - component: Forwarding
+    action: ForwardedOutOfOverlay
+capturedPacket:
+  srcIP: 10.244.1.2
+  dstIP: 142.250.182.78
+  length: 84
+  ipHeader:
+    protocol: 1
+    ttl: 64
+    flags: 2
+  tranportHeader:
+    icmp:
+      id: 29304
+      sequence: 1
+`,
+		},
+	}
+
+	for _, tt := range tcs {
+		t.Run(tt.name, func(t *testing.T) {
+			client := antreafakeclient.NewSimpleClientset()
+			getClients = func(cmd *cobra.Command) (kubernetes.Interface, antrea.Interface, error) {
+				return k8sClient, client, nil
+			}
```

Comment:
You can move line 230 `k8sClient := k8sfake.NewSimpleClientset(pod1.DeepCopyObject(), pod2.DeepCopyObject())` before these lines, and I think you can move these two assignments out of loop.